sdk/search/Azure.Search.Documents/src/Batching/Publisher.cs

```diff
@@ -488,10 +503,24 @@ protected async Task SubmitBatchAsync(IList<PublisherAction<T>> batch, Cancellat
             bool retriable = skipIncrement || action.RetryAttempts++ < MaxRetries;
             if (retriable)
             {
-                _retry.Enqueue(action);
+                EnqueueRetry(new List<PublisherAction<T>>() { action });
```

Comment:
Can we give more details about what failed?

Reply:
Not from here. This code is agnostic to reason for retrying an action.
We can add more event logs in other places.